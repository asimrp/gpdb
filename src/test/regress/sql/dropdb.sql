CREATE DATABASE db_with_dropfailed;
DROP TABLE IF EXISTS before_drop;
CREATE TABLE before_drop(oid Oid);

INSERT INTO before_drop SELECT oid FROM pg_database WHERE datname='db_with_dropfailed';

-- inject fault on content0 primary to error out before removing object
SELECT gp_inject_fault2('drop_db_before_remove_object', 'panic', dbid, hostname, port)
FROM gp_segment_configuration 
WHERE content=0 and role='p';

-- should crash on segment 0
DROP DATABASE db_with_dropfailed;

-- cleanup
SELECT gp_inject_fault2('drop_db_before_remove_object', 'reset', dbid, hostname, port)
FROM gp_segment_configuration 
WHERE content=0 AND role='p';

SELECT * FROM pg_database WHERE datname='db_with_dropfailed';

DROP TABLE before_drop;
DROP DATABASE db_with_dropfailed;
