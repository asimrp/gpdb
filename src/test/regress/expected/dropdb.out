CREATE DATABASE db_with_dropfailed;
DROP TABLE IF EXISTS before_drop;
CREATE TABLE before_drop(oid Oid);
INSERT INTO before_drop SELECT oid FROM pg_database WHERE datname='db_with_dropfailed';
-- inject fault on content0 primary to error out before removing object
SELECT gp_inject_fault2('drop_db_before_remove_object', 'panic', dbid, hostname, port)
FROM gp_segment_configuration 
WHERE content=0 and role='p';
 gp_inject_fault2 
------------------
 Success:
(1 row)

-- should crash on segment 0
DROP DATABASE db_with_dropfailed;
ERROR:  fault triggered, fault name:'drop_db_before_remove_object' fault type:'panic'  (seg0 10.34.49.130:25432 pid=27065)
-- cleanup
SELECT gp_inject_fault2('drop_db_before_remove_object', 'reset', dbid, hostname, port)
FROM gp_segment_configuration 
WHERE content=0 AND role='p';
ERROR:  connection to dbid 2 Weinan-Pivotal-Mac.local:25432 failed (gp_inject_fault.c:131)
SELECT * FROM pg_database WHERE datname='db_with_dropfailed';
      datname       | datdba | encoding | datcollate  |  datctype   | datistemplate | datallowconn | datconnlimit | datlastsysoid | datfrozenxid | datminmxid | dattablespace | datacl 
--------------------+--------+----------+-------------+-------------+---------------+--------------+--------------+---------------+--------------+------------+---------------+--------
 db_with_dropfailed |     10 |        6 | en_US.utf-8 | en_US.utf-8 | f             | t            |           -1 |         12801 |          723 |          1 |          1663 | 
(1 row)

DROP TABLE before_drop;
DROP DATABASE db_with_dropfailed;
